docs/desktop/gui/html.md

```diff
@@ -65,9 +68,107 @@ This element is used to create iframes to embed content from other sources to th
 
 Renders a [HTML5 Canvas](https://html.spec.whatwg.org/multipage/scripting.html#the-canvas-element) to the DOM. Has methods to access the render context as well to configure the dimensions of the Canvas.
 
+
+Code Sharing
+------------
+When you create instances of `qx.html.Element` on the server, at some point you can call `serialize()` to convert it into
+an HTML string which the server can pass to the client; this is useful because it provides a structured mechanism for
+defining and manipulating the DOM before it reaches the browser.
+
+However, rather than simply creating instances of `qx.html.Element`, you can derive classes from it and this has a  
+particular advantage in that you code can be used on the client and the server, and will be able to connect to the
+browser's "real" DOM nodes.
+
+For example, consider this class:
+
+```
+qx.Class.define("mypackage.MySignupForm", {
+  extend: qx.html.Element,
+  
+  construct() {
+    this.base(arguments, "div");
+    this.setCaption("Login");
+    let body = this.getBody();
+    body.add(<h1>Welcome to My Corporate Website</h1>);
+    body.add(
+        <p>This website is for internal use by authorised personnel only
+      );
+    body.add(this.getQxObject("form"));
+    body.add(
+        <p><a href="#">Click here if you have <b>forgotten your password</b></a></p>
+      );
+    this.setStyle("max-width", "600px");
+  },
+  
+  members: {
+    _createQxObjectImpl(id) {
+      switch(id) {
+      case "form":
+        let form = <form method="post" action="#"></form>;
```

Comment:
Does this require compiler support (jsx) ?

Reply:
yes - i've updated the doc